src/mds/CDir.cc

```diff
@@ -3811,10 +3811,13 @@ bool CDir::should_split_fast() const
     const CDentry *dn = p.second;
     if (!dn->get_projected_linkage()->is_null()) {
       effective_size++;
+
+      if (effective_size > fast_limit) [[unlikely]]
```

Comment:
Ummm.. I see that cut shorting the loop iteration along with using `unlikely` might just be a _bit_ beneficial.

Reply:
I don't understand this comment. Are you confirming my statement that the `[[unlikely]]` could make a tiny difference? I didn't measure the `[[unlikely]]`, but the real optimization is the shortcut which can make a big difference (5-6% CPU down to <1% CPU for this method in a `fio` benchmark creating thousands of files).

Reply:
Yes, I am in agreement with the change, but since the CPU utilisation improvement wasn't mentioned anywhere, I wasn't 100% sure if this would bring noticeable improvements. Could you mention about the CPU utilization improvement in the commit message please?

Reply:
there are no guarantees though

Reply:
> Could you mention about the CPU utilization improvement in the commit message please?

Amended.